Include/internal/pycore_signal.h

```diff
@@ -29,6 +42,77 @@ extern "C" {
 #  define Py_NSIG 64               // Use a reasonable default value
 #endif
 
+#if defined(MS_WINDOWS) && defined(SOCKET)
+#  define INVALID_FD ((SOCKET)-1)
+#else
+#  define INVALID_FD (-1)
+#endif
+
+struct _signals_runtime_state {
+    volatile struct {
+        _Py_atomic_int tripped;
+        /* func is atomic to ensure that PyErr_SetInterrupt is async-signal-safe
+         * (even though it would probably be otherwise, anyway).
+         */
+        _Py_atomic_address func;
+    } handlers[Py_NSIG];
+
+    volatile struct {
+#ifdef MS_WINDOWS
+#  ifdef SOCKET
+        SOCKET fd;
+#  else
+        // <winsock2.h> wasn't included already,
+        // we use something compatible with SOCKET.
+        int fd;
+#  endif
```

Comment:
@zooba, something about this fallback doesn't feel right.  This is what I understood from your recommendation, but I may have misunderstood.  How can I be confident about the type?  (I may even have seen a clue that it should be `volatile int`.)

Reply:
You shouldn't need any fallback here because you've already failed the build if the type isn't defined.